foursight_core/react/api/auth.py

```diff
@@ -0,0 +1,203 @@
+        i.e. list of known environments, default environment, domain, and aud (Auth0 client ID)
+        is required for the Auth0 login box (Auth0Lock) on the client-side (i.e. React UI). This
+        info is gotten from the given decoded authtoken or if not set then sets this info explicitly.
+        """
+        if authtoken_decoded:
+            response = authtoken_decoded
+        else:
+            response = {
+                "known_envs": self.envs.get_known_envs(),
+                "default_env": self.envs.get_default_env(),
+                "domain": self._get_domain(request),
+                "aud": self.auth0_client_id  # Needed for Auth0Lock login box on client-side.
+            }
+        response["authenticated"] = authenticated
+        response["authorized"] = False
+        response["status"] = status
+        return response
+
+    def _create_not_authenticated_response(self, request: dict, status: str, authtoken_decoded: dict = None) -> dict:
+        """
+        Creates a response suitable for a request which is NOT authenticated.
+        """
+        return self._create_not_authorized_response(request, status, authtoken_decoded, False)
+
+    @staticmethod
+    def _get_domain(request: dict) -> str:
+        if request:
+            try:
+                return request.get("headers", {}).get("host")
+            except:
+                pass
+        return ""
+
+    def get_aws_credentials(self, env: str) -> dict:
+        """
+        Returns basic AWS credentials info (NOT the secret).
+        This is just for informational/display purposes in the UI.
+        This has nothing to do with the rest of the authentication
+        and authorization stuff here but vaguely related so here seems fine.
+        """
+        aws_credentials = Auth.cache_aws_credentials.get(env)
+        if not aws_credentials:
+            try:
+                session = boto3.session.Session()
+                credentials = session.get_credentials()
+                access_key_id = credentials.access_key
+                region_name = session.region_name
+                caller_identity = boto3.client("sts").get_caller_identity()
+                user_arn = caller_identity["Arn"]
+                account_number = caller_identity["Account"]
+                aws_credentials = {
+                    "aws_account_number": account_number,
+                    "aws_user_arn": user_arn,
+                    "aws_access_key_id": access_key_id,
+                    "aws_region": region_name,
+                    "auth0_client_id": self.auth0_client_id
+                }
+                # Try getting the account name though probably no permission at the moment.
+                aws_account_name = None
+                try:
+                    aws_credentials["aws_account_name"] = \
+                        boto3.client('iam').list_account_aliases()['AccountAliases'][0]
+                except:
+                    pass
```

Comment:
I have a strong preference for using parentheses rather than backslash for this kind of expression runover because the meaning is the same if you remove the newline and because if you need several lines, as happens a lot in conditionals, you only have to write it on the first and last line. It makes it much easier to make small adjustments in where the line breaking is.

```suggestion
                try:
                    aws_credentials["aws_account_name"] = (
                        boto3.client('iam').list_account_aliases()['AccountAliases'][0])
                except Exception:
                    pass
```

Reply:
Got it, thanks.